platforms/database.go

```diff
@@ -0,0 +1,93 @@
+package platforms
+
+import (
+	"runtime"
+	"strings"
+)
+
+// validComboString defines valid GOOS/GOARCH combinations.
+// Should be updated from https://golang.org/doc/install/source#environment.
+// This is why we leave it as a string.
+const (
+	validComboString = `
+$GOOS	$GOARCH
+android	arm
+darwin	386
+darwin	amd64
+darwin	arm
+darwin	arm64
+dragonfly	amd64
+freebsd	386
+freebsd	amd64
+freebsd	arm
+linux	386
+linux	amd64
+linux	arm
+linux	arm64
+linux	ppc64
+linux	ppc64le
+linux	mips
+linux	mipsle
+linux	mips64
+linux	mips64le
+netbsd	386
+netbsd	amd64
+netbsd	arm
+openbsd	386
+openbsd	amd64
+openbsd	arm
+plan9	386
+plan9	amd64
+solaris	amd64
+windows	386
+windows	amd64
+`
+
+	// from https://golang.org/src/go/build/syslist.go
+	goosList   = "android darwin dragonfly freebsd linux nacl netbsd openbsd plan9 solaris windows zos "
+	goarchList = "386 amd64 amd64p32 arm armbe arm64 arm64be ppc64 ppc64le mips mipsle mips64 mips64le mips64p32 mips64p32le ppc s390 s390x sparc sparc64 "
+)
+
+func isKnownOS(os string) bool {
+	return strings.Contains(goosList, os)
+}
+
+func isKnownArch(arch string) bool {
+	return strings.Contains(goarchList, arch)
+}
+
+func normalizeOS(os string) string {
+	if os == "" {
+		return runtime.GOOS
+	}
+	os = strings.ToLower(os)
+	return os
+}
+
+// normalizeArch normalizes the architecture.
+func normalizeArch(arch, variant string) (string, string) {
+	arch = strings.ToLower(arch)
+	switch arch {
+	case "i386":
+		arch = "386"
+		variant = ""
+	case "x86_64", "x86-64":
+		arch = "amd64"
+		variant = ""
+	case "aarch64":
+		arch = "arm64"
+		variant = "" // v8 is implied
```

Comment:
`if variant == "v8" { variant = ""}`? (for better support for future non-v8 variants)